rust/src/ssh/logger.rs

```diff
@@ -16,49 +16,33 @@
  */
 
 use super::ssh::SSHTransaction;
-use crate::json::*;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
 
-fn log_ssh(tx: &SSHTransaction) -> Option<Json> {
+fn log_ssh(tx: &SSHTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
     if tx.cli_hdr.protover.len() == 0 && tx.srv_hdr.protover.len() == 0 {
-        return None;
+        return Err(JsonError::InvalidState);
```

Comment:
Is this the way to go ?

Reply:
No, the JsonError should probably only be emitted from JsonBuilder functions.  InvalidState means you tried to "append_string" or something while not in array, etc.

Reply:
ok, so changing the function prototype to return `Result<bool`

Reply:
Only use bool if that will be actually used.. Otherwise `Result<(), JsonError>` is enough to signal success/error.

Reply:
Ah, yes.. It is used, I see.